common/app/model/ShareLinks.scala

```diff
@@ -13,6 +15,19 @@ case class ShareLink (
   val userMessage: String = platform.userMessage
 }
 
+final case class ShareLinkMeta(
+  visible: Seq[ShareLink],
+  hidden: Option[NonEmptyList[ShareLink]]
```

Comment:
The idea is that there is a big difference between the empty case and the other cases. In the empty case we don't want to be doing anything. In the non-empty case we allow revealing arbitrarily many elements.

I'll drop this though as the `cats` dependency is going